components/GroupLinkSearch.js

```diff
@@ -0,0 +1,52 @@
+import { useState } from "react";
+import { Combobox } from "@headlessui/react";
+import Label from "./form/Label";
+
+function GroupLinkSearch({ selectedGroup, handleGroupSelection, groups }) {
+  const [query, setQuery] = useState("");
+
+  const handleGroupChange = (e) => {
+    setQuery(e.target.value);
+    const currentValue = e.target.value;
+    if (!groups.includes(currentValue)) {
+      handleGroupSelection(currentValue);
+    }
+  };
+
+  const filteredGroup =
+    query === ""
+      ? groups.slice(0, 5)
+      : groups
+          .filter((group) => {
+            return group.toLowerCase().includes(query.toLowerCase());
+          })
+          .slice(0, 5);
+
+  return (
+    <Combobox value={selectedGroup} onChange={handleGroupSelection}>
+      <Label htmlFor="search-groups">Group</Label>
+      <Combobox.Input
+        name="search-group"
+        onChange={handleGroupChange}
+        className={`border-2 transition-all duration-250 ease-linear rounded px-6 py-2 mb-2 block w-full dark:bg-primary-high hover:border-tertiary-medium focus:ring-0 focus:border-tertiary-medium focus:outline-0`}
+      />
```

Comment:
It would be better if we associated the `label` with the `input` field explicitly by using `id` attribute. This enhances accessibility by establishing a clear connection between `label` and `input`.

The following resources could be helpful: https://www.w3.org/WAI/tutorials/forms/labels/#associating-labels-explicitly

Reply:
If this still needs improving, could you raise as a separate issue, thank you 👍

Reply:
Sure, I'll create a new issue to address the improvement. Thanks for your guidance! 👍